File: 1003/Main.py
def pibo(piboMemo, n):
    if(piboMemo[n] is not None):
        return piboMemo[n]
    else:
        piboMemo[n] = pibo(piboMemo, n-1) + pibo(piboMemo,n-2)
        return piboMemo[n]


piboMemo = [None for _ in range(41)]
piboMemo[0] = 0
piboMemo[1] = 1

caseNumber = int(input())

for _ in range(caseNumber):
    tmp = int(input())
    if(tmp == 0):
        print(1,0)
    else:
        print(pibo(piboMemo,tmp-1),pibo(piboMemo,tmp))

File: 1003/src/Main.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
    public static int pibo(int[]arr, int n){
        arr[n] = pibo(arr,n-1) + pibo(arr,n-2);
        return 0;
    }
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        int[] arr = new int[42];
        arr[0] = 0;
        arr[1] = 1;
        int n = Integer.parseInt(bf.readLine());
        for (int i = 0; i<n; i++){
            int tmp =Integer.parseInt(bf.readLine());
            if (tmp == 0){
                System.out.println(arr[tmp-1]+" "+arr[tmp]);
            }
            else if (arr[tmp] == 0){
                pibo(arr,tmp);
                System.out.println(arr[tmp-1]+" "+arr[tmp]);
            }
            else{
                System.out.println(arr[tmp-1]+" "+arr[tmp]);
            }

        }
    }
}